src/plugins/dump/dump.cpp

```diff
@@ -476,6 +476,68 @@ int unserialise (std::istream & is, ckdb::Key * parentKey, ckdb::KeySet * ks, bo
 	return ELEKTRA_PLUGIN_STATUS_SUCCESS;
 }
 
+/**
+ * @brief An implementation of std::streambuf that is based on `FILE *` from the C API.
+ *
+ * This class allows us to read from and write to a `FILE *` via the std::iostream APIs.
+ */
+class FileStreamBuf : public std::streambuf
+{
+public:
+	FileStreamBuf (FILE * file) : file_ (file)
+	{
+	}
+
+protected:
+	std::streamsize xsputn (const char_type * s, std::streamsize n) override
+	{
+		int r = fwrite (s, 1, n, file_);
+		fflush (file_);
+		return r;
+	};
+
+	int_type overflow (int_type ch) override
+	{
+		int r = fwrite (&ch, 1, 1, file_);
+		fflush (file_);
+		return r;
+	}
+
+	int_type underflow () override
+	{
+		int c = fgetc (file_);
+		if (c == EOF)
+		{
+			this->setg (nullptr, nullptr, nullptr);
+		}
+		else
+		{
+			buf_ = c;
+			this->setg (&buf_, &buf_, &buf_ + 1);
+		}
+		return this->gptr () == this->egptr () ? std::char_traits<char>::eof () :
+							       std::char_traits<char>::to_int_type (*this->gptr ());
+	}
+
+private:
+	FILE * file_;
+	char buf_;
+};
+
+int freadks (KeySet * ks, FILE * file, Key * errorKey)
```

Comment:
rename to fserialize, funserialize (also rename serialise and unserialise)

Reply:
> also rename serialise and unserialise

Why?

Reply:
Because it is still British English and we decided to go for American English.

Reply:
Oh okay, I thought you didn't like the term serialize/unserialise for some reason